### DISCUSSION TITLE
OpenZFS fails to install on CentOS Stream 9

### DISCUSSION THREAD

**Discussion by dylanmtaylor** [openzfs/zfs#12817] [Q&A]
Are packages not being built against the latest CentOS Stream release yet? I followed the instructions for RHEL based distros, but don't think the repo is there for el9.

**Comment by Daniel-Nashed** [comment]
Did you get it working meanwhile? I have the same question.
It looks like the latest repo is 8.5.

But the project says it supports later kernels.

I thought about building it on my own meanwhile until it is available as a repository.

Does anyone have an update on it? CentOS Stream 9 has been around for a while.

Huge thanks!

Daniel Nashed

**Comment by hubick** [comment]
I just forced the 8.5 release onto RHEL 9. The zfs RPM deps didn't like the newer python, so I did `repoquery --location zfs` and `wget` to grab the rpm and `rpm -i --nodeps` on it, then `dnf install libzfs4 libzpool4 sysstat zfs-kmod --nogpgcheck` (the new version of rpm in 9 is stricter and refuses to import the ZFS repo cert now). The zfs modules built and loaded, and it imported and scrubbed my pool and 24 hrs with my home dir on it, so far so good. I did get one kernel panic immediately after zfs loaded during one boot, but not regularly, so YMMV.

**Comment by Daniel-Nashed** [comment]
thanks @hubick. This sounds a bit like an experiment. For my servers with CentOS Stream, I am staying for now on CentOS Stream 8.
For a new lab server I am looking into SUSE Leap 15.3, which has OpenZFS included in a separate repo.
I will for sure wait until it is officially available for CentOS Stream 9 and RHEL 9. 

Looks like I will have a mix of SUSE Leap and CentOS Stream now.. I figured out how to install it on my favorite provider manually on a virtual host - even they don't provide it out of the box. having different Linux distributions sounds like a good strategy anyway ;-)

**Reply by hubick** [reply]
Yeah, I'm just doing this for a ZFS mirror across two SSD's I use for my home dir on my laptop. I have backups and the main risk is my productivity this week if I have to reinstall RHEL Workstation 8.6 ;-)

**Comment by felfert** [comment]
+1
PLEASE release packages in the epel9 repo.

**Comment by felfert** [comment]
See #13738
